spec/unit/puppet/face_spec.rb

```diff
@@ -79,6 +79,21 @@
           expect(read_html(tmp, 'test', 'Puppet4xFunctions.html')).to include("This is a function which is used to test puppet strings")
         end
       end
+
+      it "should parse single-line documentation strings before a given function" do
+         using_module('test') do |tmp|
+            Dir.chdir('test')
+
+            Puppet::Face[:strings, :current].yardoc
+
+            expect(read_html(tmp,
+              'test', 'outer.html')).to include("Puppet Class: outer")
+            expect(read_html(tmp, 'test',
+              'outer/middle.html')).to include("Puppet Class: middle")
+            expect(read_html(tmp, 'test',
+              'outer/middle/inner.html')).to include("Puppet Class: inner")
```

Comment:
I think it would be worthwhile to check in the test that the classes have the expected namespaces. I believe the namespace for each class should reflect the class it's nested in.